umodbus/serial.py

```diff
@@ -58,18 +52,197 @@ def __init__(self,
+                      function_code: int,
+                      request_register_addr: int,
+                      request_register_qty: int,
+                      request_data: list,
+                      values: Optional[list] = None,
+                      signed: bool = True) -> Optional[Awaitable]:
+        """
+        Send a response to a client.
+
+        :param      slave_addr:             The slave address
+        :type       slave_addr:             int
+        :param      function_code:          The function code
+        :type       function_code:          int
+        :param      request_register_addr:  The request register address
+        :type       request_register_addr:  int
+        :param      request_register_qty:   The request register qty
+        :type       request_register_qty:   int
+        :param      request_data:           The request data
+        :type       request_data:           list
+        :param      values:                 The values
+        :type       values:                 Optional[list]
+        :param      signed:                 Indicates if signed
+        :type       signed:                 bool
+        :returns:   Request response - None for a synchronous server, or
+                    an awaitable for an asynchronous server due to AsyncRequest
+        :rtype      Awaitable, optional
+        """
+        modbus_pdu = functions.response(
+            function_code=function_code,
+            request_register_addr=request_register_addr,
+            request_register_qty=request_register_qty,
+            request_data=request_data,
+            value_list=values,
+            signed=signed
+        )
+        return self._send(modbus_pdu=modbus_pdu, slave_addr=slave_addr)
+
+    def send_exception_response(self,
+                                slave_addr: int,
+                                function_code: int,
+                                exception_code: int) -> Optional[Awaitable]:
+        """
+        Send an exception response to a client.
+
+        :param      slave_addr:      The slave address
+        :type       slave_addr:      int
+        :param      function_code:   The function code
+        :type       function_code:   int
+        :param      exception_code:  The exception code
+        :type       exception_code:  int
+        :returns:   Request response - None for a synchronous server, or
+                    an awaitable for an asynchronous server due to AsyncRequest
+        :rtype      Awaitable, optional
+        """
+        modbus_pdu = functions.exception_response(
+            function_code=function_code,
+            exception_code=exception_code)
+        return self._send(modbus_pdu=modbus_pdu, slave_addr=slave_addr)
+
+    def _send(self, modbus_pdu: bytes, slave_addr: int) -> None:
+        raise NotImplementedError("Must be overridden by subclasses")
+
+
+class RTUServer(CommonRTUFunctions, CommonModbusFunctions):
```

Comment:
⚠️ this will be a breaking change, I can understand the beauty of aligning the names, but as MicroPython boards prior to v1.19 are not able to distinglish breaking changes when using `upip`, I would vote for not renaming this now. All examples would need an update as well

Reply:
Agreed; I'll rename it back to `Serial`.

Reply:
Renamed it back to `Serial`, but I left the `AsyncRTUServer` class as is for the time being. Since I'm not sure whether renaming `Serial` to `RTUServer` would be likely in the future, should I rename it to e.g. `AsyncSerial`?